qcodes/instrument_drivers/ZI/ZIUHFLI.py

```diff
@@ -0,0 +1,1925 @@
+                       'Aux Out 2 Offset': 'auxouts/1/offset',
+                       'Aux Out 3 Offset': 'auxouts/2/offset',
+                       'Aux Out 4 Offset': 'auxouts/3/offset',
+                       'Demod 1 Phase Shift': 'demods/0/phaseshift',
+                       'Demod 2 Phase Shift': 'demods/1/phaseshift',
+                       'Demod 3 Phase Shift': 'demods/2/phaseshift',
+                       'Demod 4 Phase Shift': 'demods/3/phaseshift',
+                       'Demod 5 Phase Shift': 'demods/4/phaseshift',
+                       'Demod 6 Phase Shift': 'demods/5/phaseshift',
+                       'Demod 7 Phase Shift': 'demods/6/phaseshift',
+                       'Demod 8 Phase Shift': 'demods/7/phaseshift',
+                       'Osc 1 Frequency': 'oscs/0/freq',
+                       'Osc 2 Frequency': 'oscs/1/freq',
+                       'Output 1 Amplitude 4': 'sigouts/0/amplitudes/3',
+                       'Output 1 Offset': 'sigouts/0/offset',
+                       'Output 2 Amplitude 8': 'sigouts/1/amplitudes/7',
+                       'Output 2 Offset': 'sigouts/1/offset'
+                       }
+
+        self.add_parameter('sweeper_param',
+                           label='Parameter to sweep (sweep x-axis)',
+                           set_cmd=partial(self._sweep_setter,
+                                           'sweep/gridnode'),
+                           val_mapping=sweepparams,
+                           get_cmd=partial(self._sweep_getter,
+                                           'sweep/gridnode'),
+                           vals=vals.Enum(*list(sweepparams.keys()))
+                           )
+
+        # val_mapping for sweeper_units parameter
+        sweepunits = {'Aux Out 1 Offset': 'V',
+                      'Aux Out 2 Offset': 'V',
+                      'Aux Out 3 Offset': 'V',
+                      'Aux Out 4 Offset': 'V',
+                      'Demod 1 Phase Shift': 'degrees',
+                      'Demod 2 Phase Shift': 'degrees',
+                      'Demod 3 Phase Shift': 'degrees',
+                      'Demod 4 Phase Shift': 'degrees',
+                      'Demod 5 Phase Shift': 'degrees',
+                      'Demod 6 Phase Shift': 'degrees',
+                      'Demod 7 Phase Shift': 'degrees',
+                      'Demod 8 Phase Shift': 'degrees',
+                      'Osc 1 Frequency': 'Hz',
+                      'Osc 2 Frequency': 'Hz',
+                      'Output 1 Amplitude 4': 'V',
+                      'Output 1 Offset': 'V',
+                      'Output 2 Amplitude 8': 'V',
+                      'Output 2 Offset': 'V'
+                      }
+
+        self.add_parameter('sweeper_units',
+                           label='Units of sweep x-axis',
+                           get_cmd=self.sweeper_param.get,
+                           get_parser=lambda x:sweepunits[x])
```

Comment:
does this really really work as expected ? @WilliamHPNielsen  thinking about late binding of lambdas